File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagnamaster

import "strings"

func PreparationTime(layers []string, time int) int {
	if time == 0 {
		time = 2
	}
	return len(layers) * time
}

func Quantities(layers []string) (int, float64) {
	var noodles int = 0
	var sauce float64 = 0.0
	for i := 0 ; i < len(layers) ; i++ {
		if strings.ToLower(string(layers[i])) == "noodles"{
			noodles += 50
		}
		if strings.ToLower(string(layers[i])) == "sauce"{
			sauce += 0.2
		}
	}
	return noodles, sauce
}

func AddSecretIngredient(friend []string, mine []string) {
	mine[len(mine)-1] = string(friend[len(friend)-1])
}

func ScaleRecipe(receipe []float64, number int) []float64 {
	var newRec []float64 = []float64{}
	for i := 0; i < len(receipe); i++ {
		newRec = append(newRec, (receipe[i] * float64(number) / 2))
	}
	return newRec
}

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

export function cookingStatus(timer){
    if (timer === undefined){
        return 'You forgot to set the timer.'
    }
    return timer ? 'Not done, please wait.' : 'Lasagna is done.';
}

export function preparationTime(layers, time = 2){
    return layers.length * time;
}

export function quantities(layers){
    let noodles = 0;
    let sauce = 0;
    for(let i = 0 ; i < layers.length ; i++){
        if(layers[i] == 'noodles'){
            noodles += 50;
        }
        else if(layers[i] == 'sauce'){
            sauce += 0.2;
        }
    }
    return {
        noodles: noodles,
        sauce: sauce,
    }
}

export function addSecretIngredient(friendList, list){
    list.push(friendList[friendList.length - 1])
}

export function scaleRecipe(receipe, number){
    let newReceipe = {}
    for(let i in receipe){
        newReceipe[i] = receipe[i] * number / 2;
    }
    return newReceipe;
}